### PR TITLE
Implemented Single Sign On using Azure

### PR DESCRIPTION
**Pull request by nmcmahon-unosquare**
Added logout functionality to AuthenticationController
Added SingleSignOnController to deal specifically with Azure AD logins.
Enabled SSL for project (necessary for Azure AD logins)
Installed necessary NuGet Packages.
Added Azure config to Startup.cs
Removed Authenticated user and authentication service. These are not needed anymore.
Added ability to retrieve logged in user details from Azure in controllers.

### CODE REVIEW

File: v2/AdminCore.Services.Tests/EventServiceTests.cs [hunk 1]
```diff
@@ -45,7 +42,7 @@ public void CreateEvent_WhenCalled_SuccessfullyInsertsNewEventIntoDb()
       };
 
       // Act
-      _eventService.CreateEvent(eventDateDto, EventTypes.AnnualLeave);
+      _eventService.CreateEvent(eventDateDto, EventTypes.AnnualLeave, 1);
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.Services.Tests/EventServiceTests.cs]: Create a local variable for this employeeId

File: v2/AdminCore.Services/EventService.cs [hunk 2]
```diff
@@ -366,14 +363,14 @@ private EventDto ValidateRemainingHolidaysAndCreate(Event newEvent)
       throw new Exception("Not enough holidays to book");
     }
 
-    private Event BuildNewEvent(int employeeId, EventTypes eventTypes)
+    private Event BuildNewEvent(int employeeId)
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.Services/EventService.cs]: This overrides a previous fix, revert this

File: v2/AdminCore.Services/EventService.cs [hunk 3]
```diff
@@ -366,14 +363,14 @@ private EventDto ValidateRemainingHolidaysAndCreate(Event newEvent)
...
     {
       var newEvent = new Event
       {
         DateCreated = DateTime.Now,
         EmployeeId = employeeId,
         EventStatusId = (int)EventStatuses.AwaitingApproval,
-        EventTypeId = (int)eventTypes,
+        EventTypeId = (int)EventTypes.AnnualLeave,
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.Services/EventService.cs]: See above

File: v2/AdminCore.WebApi.Tests/Controllers/HolidayControllerTests.cs [hunk 4]
```diff
@@ -46,15 +43,14 @@ public void CreateHoliday_WhenCalled_ReturnsCreatedHoliday()
...
-      _authenticatedUser.RetrieveUserId().Returns(_authenticatedUser.RetrieveUserId());
 
       // Act
       var result = _controller.CreateHoliday(createViewModel);
 
       // Assert
 
       VerifyActionResult(result);
-      _eventService.Received(1).CreateEvent(Arg.Any<EventDateDto>(), EventTypes.AnnualLeave);
+      _eventService.Received(1).CreateEvent(Arg.Any<EventDateDto>(), EventTypes.AnnualLeave, 1);
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.WebApi.Tests/Controllers/HolidayControllerTests.cs]: See above, local variable for this

**Review by Jamie-Unosquare — CHANGES_REQUESTED** [review]
Biggest issue is reverting the eventtypes

File: v2/AdminCore.Services.Tests/EventServiceTests.cs (hunk 1 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.Services.Tests/EventServiceTests.cs]: Fixed

File: v2/AdminCore.Services/EventService.cs (hunk 2 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.Services/EventService.cs]: Fixed

File: v2/AdminCore.Services/EventService.cs (hunk 3 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.Services/EventService.cs]: Fixed

File: v2/AdminCore.WebApi.Tests/Controllers/HolidayControllerTests.cs (hunk 4 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.WebApi.Tests/Controllers/HolidayControllerTests.cs]: Fixed

**Review by Jamie-Unosquare — APPROVED** [review]

File: v2/AdminCore.Services/AuthenticatedUser.cs [hunk 5]
```diff
@@ -1,22 +0,0 @@
...
-    private readonly IHttpContextAccessor _httpContentAccessor;
-
-    public AuthenticatedUser(IHttpContextAccessor httpContextAccessor)
-    {
-      _httpContentAccessor = httpContextAccessor;
-    }
-
-    public int RetrieveUserId()
-    {
-      var employeeId = _httpContentAccessor.HttpContext.User.FindFirst(ClaimTypes.Name).Value;
```
Comment (BrownMark) [inline v2/AdminCore.Services/AuthenticatedUser.cs:18]: Can you give me a shout about this 👍 . Just want to check something

**Pull request merged**